### DISCUSSION TITLE
Data layer w/ Repository pattern

### DISCUSSION THREAD

**Discussion by ghost** [vueuse/vueuse#1696] [Q&A]
Hello,

I am quite new with Vue 3 and I am currently migrating a Nuxt 2 application to Nuxt Bridge. I have discovered the Composition API, composable and VueUse. I will not use Typescript for the moment because I have quite a lot to migrate. Nevertheless I would like to create a data layer based on Repository Pattern (like described in this article: https://blog.lichter.io/posts/nuxt-api-call-organization-and-decoupling/). Our application is currently using axios and useAxios from VueUse is quite useful.

If someone have an example with this kind of architecture in Nuxt Bridge or 3, I would be happy to receive any ideas or propositions!

To sum up:
- need to migrate a Nuxt 2 app to Nuxt Bridge by keep using plain javascript,
- need to consume a REST API by using Axios - and useAxios() composable,
- would like to follow the Repository Pattern or build a service data layer.

Have a good day!
Best Regards,

**Comment by camipe** [comment]
@sebastien-metzger-impact-positif Did you figure out a nice solution for this?
